requestid/metadata_annotator.go

```diff
@@ -0,0 +1,20 @@
+package requestid
+
+import (
+	"context"
+	"net/http"
+
+	"google.golang.org/grpc/metadata"
+)
+
+func NewRequestIDAnnotator() func(context.Context, *http.Request) metadata.MD {
+	return func(ctx context.Context, req *http.Request) metadata.MD {
+		if req.Header.Get(DefaultRequestIDKey) != "" {
```

Comment:
Do we need to check for `X-Request-ID` as well here?  Because by defualt Nginx Doesnt log `Request-ID` now. It logs only `X-Request-ID`